cmake/OpenCVGenConfig.cmake

```diff
@@ -128,8 +128,11 @@ endif()
 #  Part 3/3: ${BIN_DIR}/win-install/OpenCVConfig.cmake  -> For use within binary installers/packages
 # --------------------------------------------------------------------------------------------
 if(WIN32)
-  set(OpenCV_INCLUDE_DIRS_CONFIGCMAKE "\"\${OpenCV_CONFIG_PATH}/include\" \"\${OpenCV_CONFIG_PATH}/include/opencv\"")
-  set(OpenCV2_INCLUDE_DIRS_CONFIGCMAKE "\"\"")
+
+  if (CMAKE_HOST_SYSTEM_NAME MATCHES Windows)
+    set(OpenCV_INCLUDE_DIRS_CONFIGCMAKE "\"\${OpenCV_CONFIG_PATH}/include\" \"\${OpenCV_CONFIG_PATH}/include/opencv\"")
+    set(OpenCV2_INCLUDE_DIRS_CONFIGCMAKE "\"\"")
+  endif ()
```

Comment:
Thanks!
There is #5980 which should handle this case (and case below) without a magic of "if"s.

About a fix above with LIB_SUFFIX (same as in #5940): it changes current behavior, so I need to ask our teams who uses the current "share/OpenCV" way.